tests/tape_tests/test_ensemble.py

```diff
@@ -723,46 +723,84 @@ def test_update_column_map(dask_client):
     assert cmap_2.map["provenance_col"] == "p"
 
 
-def test_sync_tables(parquet_ensemble):
+@pytest.mark.parametrize("legacy", [True, False])
+def test_sync_tables(parquet_ensemble, legacy):
     """
-    Test that _sync_tables works as expected
+    Test that _sync_tables works as expected, using Ensemble-level APIs
+    when `legacy` is `True`, and EsnembleFrame APIs when `legacy` is `False`.
     """
-
-    assert len(parquet_ensemble.compute("object")) == 15
-    assert len(parquet_ensemble.compute("source")) == 2000
+    if legacy:
+        assert len(parquet_ensemble.compute("object")) == 15
+        assert len(parquet_ensemble.compute("source")) == 2000
+    else:
+        assert len(parquet_ensemble.object.compute()) == 15
```

Comment:
What would happen if I do `parquet_ensemble.object.shape[0].compute()` instead? Do I understand correctly that our overloaded version of `compute()` will not be run?

Reply:
As discussed we will not be able to detect this here and there are other cases we would like to be able to handle as well.

Filed an issue for this in
https://github.com/lincc-frameworks/tape/issues/296